### ISSUE TITLE
Nollup does not catch plugin error on change

### ISSUE THREAD

**Issue by hanswilw** [PepsRyuu/nollup#128]
When using the typescript plugin it looks like changes are handled differently from the initial build. If typescript throws, it propagates to the PluginContext and nollup crashes if it is rebundling on change – not initially.

Here I deliberately triggered a TS error, then fixed it (giving Compiled in 1404ms), then triggered it again – which throws and logs the whole error object from TS. Havent been able to dig too much into the catching logic here yet.

[Example diff with typescript support](https://github.com/hanswilw/nollup/commit/febda7b442eecf8014c7d5854d7cdb476c5f53fb)

```
$ npm start                                                                                                                                                            1 ↵  ✹ ✭master  [14:10:36]

> @ start ~/nollup/examples/example-preact
> cross-env NODE_ENV=development node ../../lib/cli.js -c --hot --content-base public --port 9001

[Nollup] Listening on http://localhost:9001
@rollup/plugin-typescript TS7026: JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists.
[Nollup] Compiled in 1401ms.

~/nollup/lib/impl/PluginContext.js:150
                throw e;
                ^
{
  pluginCode: 'TS7026',
  message: "@rollup/plugin-typescript TS7026: JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists.",
  loc: {
    column: 5,
    line: 7,
    file: '~/nollup/examples/example-preact/src/App.tsx'
  },
  frame: '\n\u001b[7m7\u001b[0m     <div class="App">\n\u001b[7m \u001b[0m \u001b[91m    ~~~~~~~~~~~~~~~~~\u001b[0m\n'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ start: `cross-env NODE_ENV=development node ../../lib/cli.js -c --hot --content-base public --port 9001`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by PepsRyuu** [comment]
Thanks for the repro! That is very peculiar. Off the top of my head I have no idea what's going on, because all errors should be caught by the ```try/catch``` in the ```dev-middleware.js``` file which works as intended for the first build.

**Comment by PepsRyuu** [comment]
I might be wrong on this theory, because I'm not very familiar with TypeScript, but it looks like the plugin instantiates a file watcher within TypeScript itself, and when Nollup tries to load a file, it waits for TypeScript to finish before responding with the TypeScript output. The watch helper that's used to make it easier to detect changes creates a new promise chain that is completely disconnected from the current promise chain. So when ```this.error``` is used inside that promise chain, it's outside of the scope of the Nollup promise chain, so it's treated as an ```uncaughtException```.

```
ts.createWatchProgram(createWatchHost(ts, context, options)); // context is PluginContext with error

async load () {
   ...
   await watchProgramHelper.wait();
   const output = findTypescriptOutput(ts, parsedOptions, id, emittedFiles);
}
```

I can add code that will respond to ```uncaughtException``` but that feels like the wrong solution and probably would have unintended consequences since there's unfulfilled bundle promises in memory. What needs to happen is that when ```this.error``` is called, it somehow tells the current build to stop. Not sure how that would be done yet.

**Comment by PepsRyuu** [comment]
https://github.com/PepsRyuu/nollup/pull/129 Prototype. Still going to require a bit more work, not satisfied with it fully yet but it seems to work as far as I can tell.

**Comment by hanswilw** [comment]
Great! Really appreciate the prompt investigation! 👍 I was able to bypass it in the meanwhile by using `noEmitOnError: false`, so it just warns but still emits.

**Comment by PepsRyuu** [comment]
Fixed merge and will be released in ```0.13.5```! :)
